src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -78,6 +78,91 @@ public static string CreateClientId(string prefix)
             return $"VS ({prefix}) ({Guid.NewGuid().ToString()})";
         }
 
+        public static Task<RemoteHostClient?> TryGetClientAsync(Project project, CancellationToken cancellationToken)
+        {
+            if (!RemoteSupportedLanguages.IsSupported(project.Language))
+            {
+                return SpecializedTasks.Default<RemoteHostClient?>();
+            }
+
+            return TryGetClientAsync(project.Solution.Workspace, cancellationToken);
+        }
+
+        public static Task<RemoteHostClient?> TryGetClientAsync(Workspace workspace, CancellationToken cancellationToken)
+        {
+            var service = workspace.Services.GetService<IRemoteHostClientService>();
+            if (service == null)
+            {
+                return SpecializedTasks.Default<RemoteHostClient?>();
+            }
+
+            return service.TryGetRemoteHostClientAsync(cancellationToken);
+        }
+
+        /// <summary>
+        /// Creates <see cref="SessionWithSolution"/> for the <paramref name="serviceName"/> if possible, otherwise returns <see langword="null"/>.
+        /// </summary>
+        public async Task<SessionWithSolution?> TryCreateSessionAsync(string serviceName, Solution solution, object? callbackTarget, CancellationToken cancellationToken)
+        {
+            var connection = await TryCreateConnectionAsync(serviceName, callbackTarget, cancellationToken).ConfigureAwait(false);
+            if (connection == null)
+            {
+                return null;
+            }
+
+            return await SessionWithSolution.CreateAsync(connection, solution, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Creates <see cref="KeepAliveSession"/> for the <paramref name="serviceName"/>, otherwise returns <see langword="null"/>.
+        /// </summary>
+        public async Task<KeepAliveSession?> TryCreateKeepAliveSessionAsync(string serviceName, object? callbackTarget, CancellationToken cancellationToken)
+        {
+            var connection = await TryCreateConnectionAsync(serviceName, callbackTarget, cancellationToken).ConfigureAwait(false);
+            if (connection == null)
+            {
+                return null;
+            }
+
+            return new KeepAliveSession(this, connection, serviceName, callbackTarget);
+        }
+
+        public async Task<bool> TryRunRemoteAsync(string serviceName, string targetName, Solution solution,
+            IReadOnlyList<object> arguments, object? callbackTarget, CancellationToken cancellationToken)
```

Comment:
do we need this version, when we have the `TryRunREmoteAsync<T>` version below?

Reply:
Yes, this is for T = void.